app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -171,10 +171,15 @@ def setup_agent(instance)
     # prepare work directory
     ssh.perform_commands(["sudo mkdir -p #{WORK_DIR}"])
     ssh.perform_commands(["sudo chmod go+w #{WORK_DIR}"])
+
     # scp the default setting yaml file
-    Tempfile.open('config.yml') do |config|
+    config = Tempfile.new('config.yml')
+    begin
       config.write(create_config_yaml)
-      scp_file(ip, agent_ami_login_user, auth_key, config, WORK_DIR)
+      config.close
+      out = scp_file(ip, agent_ami_login_user, auth_key, config.path, "#{WORK_DIR}/config.yml")
```

Comment:
It seems there is no need to assign the return value to `out`.

Reply:
Yes, forgot to remove it. I'll change it in next PR.